### PR TITLE
Feat/additional paths

### PR DESCRIPTION
**Pull request by mayki0909**
https://app.clickup.com/t/2e5ww7u

### CODE REVIEW

File: src/stake/stake.service.ts [hunk 1]
```diff
@@ -81,6 +109,45 @@ export class StakeService {
...
+    let totalAmount = 0
+    let totalFarmed = 0
+
+    stakes.map(stake => {
+      if(!members.includes(stake.wallet)) members.push(stake.wallet)
+      totalAmount += stake.amount
+      averageAPY += stake.plan?.apy
+      totalFarmed += stake.farmed
+    })
+    averageAPY = averageAPY / stakes.length
```
Comment (matjaz-NFT) [inline src/stake/stake.service.ts:136]: can happen that stakes length is 0?

File: src/stake/stake.service.ts [hunk 2]
```diff
@@ -81,6 +109,45 @@ export class StakeService {
     return stakes
   }
 
+  async activeStakesPool(poolName: string, wallet: string): Promise<ActiveStakesPool> {
+    const pool = await this.poolService.findOneByHandle(poolName)
```
Comment (matjaz-NFT) [inline src/stake/stake.service.ts:113]: what if pool is not found/doesn't exist anymore?

File: src/stake/stake.service.ts [hunk 3]
```diff
@@ -81,6 +109,45 @@ export class StakeService {
...
+    stakes.map(stake => {
+      if(!members.includes(stake.wallet)) members.push(stake.wallet)
+      totalAmount += stake.amount
+      averageAPY += stake.plan?.apy
+      totalFarmed += stake.farmed
+    })
+    averageAPY = averageAPY / stakes.length
+
+    return {
+      social: poolName.split('-')[0],
```
Comment (matjaz-NFT) [inline src/stake/stake.service.ts:139]: poolName is input of function, which is not safe to use. here you could use pool object to get pool name instead?

File: src/stake/stake.service.ts [hunk 4]
```diff
@@ -71,6 +74,31 @@ export class StakeService {
...
+      collected: false,
+    }).populate('pool')
+
+    const distribution = {}
+
+    for (const stake of stakes) {
+      const poolSplit = stake.pool.creator.split('-')
+      if (poolSplit.length == 2) {
+        const chanel = poolSplit[0]
+        if (distribution[chanel]) distribution[chanel] += stake.amount
```
Comment (matjaz-NFT) [inline src/stake/stake.service.ts:89]: you could use if case to set distribution[chanel] to 0 first time, then you can always run `distribution[chanel] += stake.amount` after

File: src/stake/stake.service.ts [hunk 5]
```diff
@@ -71,6 +74,31 @@ export class StakeService {
...
+        else distribution[chanel] = stake.amount
+      }
+    }
+
+    const distributionArray = []
+    for (const chanel in distribution) {
+      distributionArray.push({chanel, distribution: (distribution[chanel]/totalStaked). toFixed(3)})
```
Comment (matjaz-NFT) [inline src/stake/stake.service.ts:96]: possible division with 0?

**Review by matjaz-NFT — CHANGES_REQUESTED** [review]

File: src/stake/stake.service.ts (hunk 1 above)
Comment (mayki0909) [reply src/stake/stake.service.ts:136]: Yes it can, but it doesn't crash app
image

File: src/stake/stake.service.ts (hunk 2 above)
Comment (mayki0909) [reply src/stake/stake.service.ts:113]: It returns null response as shown on screenshot above

File: src/stake/stake.service.ts (hunk 3 above)
Comment (mayki0909) [reply src/stake/stake.service.ts:139]: Okey

File: src/stake/stake.service.ts (hunk 1 above)
Comment (matjaz-NFT) [reply src/stake/stake.service.ts:137]: shouln't we return non 200 response, if pool is not found?

**Review by matjaz-NFT — APPROVED** [review]

File: src/stake/stake.service.ts [hunk 6]
```diff
@@ -71,6 +75,31 @@ export class StakeService {
...
+    const stakes: Stake[] = await this.stakeModel.find({
+      collected: false,
+    }).populate('pool')
+
+    const distribution = {}
+
+    for (const stake of stakes) {
+      const poolSplit = stake.pool.creator.split('-')
+      if (poolSplit.length == 2) {
+        const chanel = poolSplit[0]
```
Comment (mrVers) [inline src/stake/stake.service.ts:89]: Typo. `chanel` -> `channel`

File: src/stake/stake.service.ts (hunk 1 above)
Comment (mayki0909) [reply src/stake/stake.service.ts:162]: Now it return 400 not found

**Review by mrVers — APPROVED** [review]

**Pull request merged**